OCAExplorer/src/components/Demo.tsx

```diff
@@ -90,24 +90,31 @@ const demoStates = {
   }]
 }
 
-export type DemoState = "NotRunning" | "RunningIntro" | "RunningBranding" | "RunningAll"
+export enum DemoState{
+    NotRunning,
+    RunningIntro,
+    RunningBranding,
+    RunningAll,
+    SeenDemo,
+}
 
-export function Demo({runDemo, theme, resetFunc }: {runDemo: DemoState, theme: Theme, resetFunc: () => void }) {
+export function Demo({runDemo, theme, resetFunc, skipFunc, }: {runDemo: DemoState, theme: Theme, resetFunc: () => void, skipFunc: () => void  }) {
   const handleJoyrideCallback = (data: any) => {
     const { status, type } = data;
-    const finishedStatuses: string[] = [STATUS.FINISHED, STATUS.SKIPPED];
 
-    if (finishedStatuses.includes(status)) {
+    if (status == STATUS.FINISHED) {
```

Comment:
Can there just be one function that runs when the demo is finished or skipped? Both will set the demo state to `SeenDemo`

If the user force starts the demo again (by clicking the info icon) then the state is reset to `NotSeenDemo`

Reply:
The only reason I have 2 branches is because we need to pause before the overlay is shown. For this reason the reset function is actually dependent on if the overlay is shown. If someone finishes the demo with the overlay shown then they have `SeenDemo` if they have not displayed the overlay then the state is `PausedDemo` to comply with the main issue posed of skip including the overlay demo even when it is not shown I have a dedicated skip function that will be used to set the state to `SeenDemo` no matter what

Reply:
That makes sense. Thanks for the clarification.